src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -31,7 +39,24 @@
 @WebServlet("/events/*")
 public class EventServlet extends HttpServlet {
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String pathName = request.getPathInfo();
+    UserService userService = UserServiceFactory.getUserService();
+    if (pathName.length() <= 1) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return ;
+    }
+
+    if (pathName.contains("gcalendar")) {
```

Comment:
This would pass for /events/notgcalendar as well. Might worth doing parsing to see if the string between / and ? matches gcalendar instead.

Reply:
Solved it